ui/src/routes/incarnations/Form.tsx

```diff
@@ -49,20 +49,69 @@ const DeleteIncarnationLink = styled.span`
   text-decoration: underline;
 `
 
+type DiffChanges = {
+  added: number,
+  removed: number
+}
+
 type FormProps = {
   mutation: (data: IncarnationInput) => Promise<IncarnationApiView>,
   deleteIncarnation?: () => Promise<void>,
   defaultValues: IncarnationInput,
+  diffChanges?: DiffChanges
   isEdit?: boolean,
   incarnationMergeRequestStatus?: MergeRequestStatus | null,
   mergeRequestUrl?: string | null,
   commitUrl?: string
   templateDataFull?: Record<string, never>
 }
 
+type ChangeSquareProps = {
+  color: 'green' | 'red' | 'gray';
+};
+
+enum ChangeColor {
+  GREEN = '#2c9c69',
+  RED = '#d73a49',
+  GRAY = '#6a737d'
+}
```

Comment:
nitpick - not a fan of `enum`. I would prefer an object literal with `as const`.